hoss/coordinate_utilities.py

```diff
@@ -179,42 +184,225 @@ def get_1D_dim_array_data_from_dimvalues(
         # Creates an entire array of `True` values.
-        is_not_fill = np.ones_like(coordinate_row_col, dtype=bool)
+        is_not_fill = np.ones_like(lat_lon_array, dtype=bool)
 
     if coordinate.is_longitude():
-        valid_indices = np.where(
-            np.logical_and(
-                is_not_fill,
-                np.logical_and(
-                    coordinate_row_col >= -180.0, coordinate_row_col <= 360.0
-                ),
-            )
-        )[0]
+        valid_indices = np.logical_and(
+            is_not_fill,
+            np.logical_and(lat_lon_array >= -180.0, lat_lon_array <= 360.0),
+        )
     elif coordinate.is_latitude():
-        valid_indices = np.where(
-            np.logical_and(
-                is_not_fill,
-                np.logical_and(coordinate_row_col >= -90.0, coordinate_row_col <= 90.0),
-            )
-        )[0]
+        valid_indices = np.logical_and(
+            is_not_fill,
+            np.logical_and(lat_lon_array >= -90.0, lat_lon_array <= 90.0),
+        )
     else:
         valid_indices = np.empty((0, 0))
 
-    return valid_indices
+    return valid_indices  # throw an exception
+
+
+def get_row_col_geo_grid_points(
+    lat_arr: np.ndarray,
+    lon_arr: np.ndarray,
+    latitude_coordinate: VariableFromDmr,
+    longitude_coordinate: VariableFromDmr,
+    # ) -> tuple[dict, dict]:
+) -> tuple[list, list]:
+    """
+    This method is used to return two valid lat lon points from a 2D
+    coordinate dataset. It gets the row and column of the latitude and longitude
+    arrays to get two valid points. This does a check for fill values and
+    This method does not go down to the next row and col. if the selected row and
+    column all have fills, it will raise an exception in those cases.
+    """
+
+    geo_grid_indices_row, geo_grid_indices_col = get_row_col_valid_indices_in_dataset(
+        lat_arr, lon_arr, latitude_coordinate, longitude_coordinate
+    )
+
+    geo_grid_col_points = [
+        (lon_arr[row, col], lat_arr[row, col]) for row, col in geo_grid_indices_row
+    ]
+    geo_grid_row_points = [
+        (lon_arr[row, col], lat_arr[row, col]) for row, col in geo_grid_indices_col
+    ]
+
+    return geo_grid_row_points, geo_grid_col_points
+
+
+def get_x_y_values_from_geographic_points(
+    points: list[tuple],  # list of data points as tuple in (lon, lat) order
+    crs: CRS,  # CRS object from PyProj
+) -> list[tuple]:  # list of X-Y points as tuple in (X, Y) order
+    """Take an input list of (longitude, latitude) coordinates and project
+    those points to the target grid. Then return the x-y dimscales
+
+    """
+    point_longitudes, point_latitudes = zip(*list(points))
+
+    from_geo_transformer = Transformer.from_crs(4326, crs)
+    points_x, points_y = (  # pylint: disable=unpacking-non-sequence
+        from_geo_transformer.transform(point_latitudes, point_longitudes)
+    )
+    x_y_points = list(zip(points_x, points_y))
+
+    return x_y_points
```

Comment:
You can avoid declaring a variable for a single immediate use and just do:

```
return list(zip(points_x, points))
```

Reply:
done. - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/1c2922126c8ebe8472e8e1bcad97a058d69100e2